tests/routes/test_expressions.py

```diff
@@ -0,0 +1,43 @@
+from django.test import TestCase
+
+from conman.routes.expressions import CharCount
+from conman.routes.models import Route
+
+from .factories import RouteFactory
+
+
+class TestCharCount(TestCase):
+    """Tests for CharCount."""
+    def test_query(self):
+        """Match the exact value of the generated query."""
+        # The "only" here is handy to keep the query as short as possible.
+        qs = Route.objects.only('id').annotate(level=CharCount('url', char='/'))
+        # Excuse the line wrapping here -- I wasn't sure of a nice way to do it.
+        # I decided it was better to just keep it simple.
+        expected = (
+            'SELECT "routes_route"."id", ' +
+            'CHAR_LENGTH("routes_route"."url") - ' +
+            'CHAR_LENGTH(REPLACE("routes_route"."url", \'/\', \'\')) AS "level" ' +
+            'FROM "routes_route"'
+        )
+        self.assertEqual(str(qs.query), expected)
+
+    def test_annotation(self):
+        """Test the expression can be used for annotation."""
+        RouteFactory.create(url='/sixth/level/path/including/root/')
+        route = Route.objects.annotate(level=CharCount('url', char='/')).get()
+        self.assertEqual(route.level, 6)  # The number of "/" in the path.
+
+    def test_calling_format(self):
+        """Ensure the 'char' argument is always a keyword-arg."""
+        with self.assertRaises(TypeError):
+            CharCount('url', 'unacceptable')
```

Comment:
It might be cleaner to use a single character here. so we're violating only one assumption of the api.